packages/e2e-test-utils/src/publishPost.js

```diff
@@ -45,17 +45,24 @@ async function publishPost() {
 
   if (prePublishChecksEnabled) {
     await openPublishPanel();
+
+    // Wait for the panel to fully slide in.
+    await page.waitForTimeout(500);
   }
 
   // Publish the post
   await page.click('.editor-post-publish-button');
 
   // Wait until the selector returns a truthy value.
-  await page.waitForFunction(() =>
-    wp.data.select('core/editor').getPermalink()
+  await page.waitForFunction(
+    () =>
+      wp.data.select('core/editor').getEditedPostAttribute('status') ===
+      'publish'
   );
```

Comment:
Can you give some more details here?

Reply:
Previously this util just checked for `wp.data.select('core/editor').getPermalink()` and that it doesn't return `null`. If it returns a truthy value (e.g. the permalink string), we assumed the post was published.

But apparently nowadays `wp.data.select('core/editor').getPermalink()` always returns a string with the permalink, even if the post is still a draft.

We now wait for the post status to actually change and for the "View Post" link to appear, so that we are more certain that the post was actually published.